lib/datadog/tracing/contrib/http/configuration/settings.rb

```diff
@@ -35,10 +41,26 @@ class Settings < Contrib::Configuration::Settings
             end
 
             option :error_status_codes do |o|
```

Comment:
I haven't add the `type` to the `error_status_code` option, because in the tests the values used are Array and Range. I didn't wanted to change the backend logic for `:array` on this PR. So I will leave the decision to adding the `type` to the original author of this integration